code/clue.js

```diff
@@ -1,59 +1,374 @@
+const weapons = [
+  rope,
+  knife,
+  candelstick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
 
-const rooms = []
+const rooms = [
+  diningRoom, 
+  conservatory, 
+  kitchen, 
+  study, 
+  library, 
+  billiardRoom, 
+  lounge, 
+  ballroom, 
+  hall, 
+  spa, 
+  livingRoom, 
+  observatory, 
+  theater, 
+  guestHouse, 
+  patio
+]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//object that keeps mystery 
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+};
 
-// This function will be invoked when you click on the killer card.
+// This function will be invoked when you click the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  if (!killerIsPicked) {
+    document.getElementById('loaderKiller').style.opacity = 1
+    setTimeout(function () {
+      document.getElementById('loaderKiller').style.opacity = 0
+      shuffleFavoriteWeapon();
+      mystery.killer = randomSelector(suspects)
+      document.getElementById('killerName').style.background = mystery.killer.color
+      document.getElementById(
+        'killerName'
+      ).innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+      document.getElementById('killerAge').innerHTML = (`Age: ${mystery.killer.age}`)
+      document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+      document.getElementById('killerDescription').innerHTML = `Descripton: ${mystery.killer.description}`
+      document.getElementById('killerFavouriteWeapon').innerHTML = `Preferred weapon: ${mystery.killer.favouriteWeapon}` //red level to check that the fav-weap changes by using function "shuffleFavouriteWeapon"
+      document.getElementById('killerImage').src = mystery.killer.image;
+      document.getElementById('killerImage').style.opacity = 1;
+      killerIsPicked = true;
+    }, 2500)
+  }
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+// This function will be invoked when you click the weapon card.
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('loaderWeapon').style.opacity = 1
+    setTimeout(function() {
```

Comment:
`setTimeout(() => {...}, 2500)` would be better approach, we shouldn't use `function` keyword a lot

Reply:
Correct, thank you. I have changed that no to arrow function instead.